app/services/api/record_laa_reference.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Api
+  class RecordLaaReference < ApplicationService
+    include CommonPlatformConnection
+
+    # rubocop:disable Metrics/ParameterLists
+    def initialize(laa_reference_id:, prosecution_case_id:, defendant_id:, offence_id:, status_code:, application_reference:, status_date:)
+      @prosecution_case_id = prosecution_case_id
+      @defendant_id = defendant_id
+      @offence_id = offence_id
+      @status_code = status_code
+      @application_reference = application_reference
+      @status_date = status_date
+      @url = "/prosecutionCases/laaReference/#{laa_reference_id}"
+    end
+    # rubocop:enable Metrics/ParameterLists
+
+    def call
+      common_platform_connection.put(url, request_body)
+    end
+
+    private
+
+    def request_body
+      {
+        prosecutionCaseId: prosecution_case_id,
+        defendantId: defendant_id,
+        offenceId: offence_id,
+        statusCode: status_code,
+        applicationReference: application_reference,
+        statusDate: status_date
+      }
+    end
+
+    attr_reader :url, :prosecution_case_id, :defendant_id, :offence_id, :status_code, :application_reference, :status_date, :response
+  end
```

Comment:
Where is this response variable being declared? I get that the `CommonPlatformConnection` has a response - but I can't see how this would be accessible to an `attr_reader` on `RecordLaaReference`.

Reply:
Good catch, this isn't needed/used here at all. Removed it.